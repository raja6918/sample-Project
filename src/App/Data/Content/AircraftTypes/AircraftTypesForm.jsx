import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PerfectScrollbar from 'react-perfect-scrollbar';
import 'react-perfect-scrollbar/dist/css/styles.css';

import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import MUIFormControl from '@material-ui/core/FormControl';

import InputSelector from '../../../../components/InputSelector';
import ErrorMessage from '../../../../components/FormValidation/ErrorMessage';
import Form from '../../../../components/FormDrawer/Form';
import FormHeader from '../../../../components/FormDrawer/FormHeader';
import FormBody from '../../../../components/FormDrawer/FormBody';
import StandardCrewComplement from '../../../../components/StandardCrewComplement';

import { prepareRestFacilities, prepareModels } from './Constants';
import { getInlineErrorMessage } from '../../../../_shared/helpers';
import {
  perfectScrollConfig,
  getFormOkButton,
  getFormHeading,
  getFormCancelButton,
} from '../../../../utils/common';

const DEFAULT_REST_FACILITY_CODE = 'NONE';

const Input = styled(TextField)`
  width: 100%;
  margin: 0;
  &.error > label {
    color: #d10000 !important;
  }
`;
const FormControl = styled(MUIFormControl)`
  width: 100%;
`;

class AircraftTypesForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      aircraftName: '',
      aircraftType: '',
      aircraftModel: '',
      restFacility: DEFAULT_REST_FACILITY_CODE,
      crewComposition: null,
      errors: {
        aircraftName: false,
        aircraftType: false,
      },
      isFormDirty: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.isOpen) {
      this.setState({
        aircraftName: nextProps.aircraft ? nextProps.aircraft.name : '',
        aircraftType: nextProps.aircraft ? nextProps.aircraft.code : '',
        aircraftModel: nextProps.aircraft ? nextProps.aircraft.modelCode : '',
        crewComposition: nextProps.aircraft
          ? nextProps.aircraft.crewComposition
          : null,
        restFacility: nextProps.aircraft
          ? nextProps.aircraft.restFacilityCode
          : DEFAULT_REST_FACILITY_CODE,
        isFormDirty: false,
        errors: {
          aircraftName: false,
          aircraftType: false,
        },
      });
    }

    if (nextProps.inlineError) {
      this.setState(nextProps.inlineError, this.checkIfFormisReady);
    }
  }

  isReadyToSubmit = () => {
    return !this.state.isFormDirty;
  };

  checkIfFormisReady = () => {
    const node_aircraftType = document.getElementById('aircraftType');
    const aircraftType = node_aircraftType
      ? node_aircraftType.value.trim()
      : '';
    const { errors, aircraftName, aircraftModel, isFormDirty } = this.state;
    const isError = errors.aircraftName || errors.aircraftType;

    if (
      aircraftType !== '' &&
      aircraftName !== '' &&
      aircraftModel !== null &&
      aircraftModel !== '' &&
      !isError
    ) {
      if (!isFormDirty) {
        this.setState({
          isFormDirty: true,
        });
      }
    } else {
      if (isFormDirty) {
        this.setState({
          isFormDirty: false,
        });
      }
    }
  };

  onFormChange = () => {
    this.checkIfFormisReady();
  };

  evaluateRegex = (regex, term) => {
    const reg = new RegExp(regex);
    const isValidRegex = reg.test(term);
    return isValidRegex;
  };

  updateErrors = (e, regexp: '') => {
    const event = e.target || e.srcElement;
    const key = event.name;
    const value = event.value;
    const isError = value === '' ? false : !this.evaluateRegex(regexp, value);

    return {
      ...this.state.errors,
      [key]: isError,
    };
  };

  handleCrewComplements = crewComposition => {
    this.setState(
      { crewComposition },
      this.props.handleCrewComplements(crewComposition)
    );
    if (crewComposition !== this.state.crewComposition) {
      this.onFormChange();
    }
  };

  onSelectChange = (target, value) => {
    this.setState(
      {
        [target]: value,
      },
      this.onFormChange
    );
  };

  handleChange = (e, regexp: '') => {
    const { inlineError } = this.props;
    const errors = this.updateErrors(e, regexp);
    const inlineErrorMessage = getInlineErrorMessage(e, inlineError);

    this.setState(
      {
        errors,
        [e.target.name]: e.target.value,
        inlineErrorMessage,
      },
      this.checkIfFormisReady
    );
  };

  render() {
    const {
      aircraft,
      openItemId,
      formId,
      models,
      restFacilities,
      t,
      readOnly,
      enableReadOnly,
      ...rest
    } = this.props;
    const {
      errors,
      aircraftType,
      aircraftName,
      aircraftModel,
      restFacility,
      crewComposition,
      inlineErrorMessage,
    } = this.state;
    const readOnlyStatus = readOnly || enableReadOnly;
    const isDisabled = readOnlyStatus ? false : this.isReadyToSubmit();
    const aircraftOp = 'DATA.aircraft.form';
    const sectionGeneral = 'DATA.aircraft.form.section.general';
    const sectionCrewComplement = 'DATA.aircraft.form.section.crewComplement';
    const errorMsg = 'ERRORS.AIRCRAFT';

    let codeErrorMessage = t(`${errorMsg}.aircraftType`);

    if (inlineErrorMessage) {
      codeErrorMessage = inlineErrorMessage;
    }

    return (
      <Form
        okButton={getFormOkButton(t, aircraft)}
        cancelButton={getFormCancelButton(t, readOnlyStatus)}
        isDisabled={isDisabled}
        onChange={this.onFormChange}
        anchor={'right'}
        enableReadOnly={enableReadOnly}
        {...rest}
      >
        <FormHeader>
          <span>{getFormHeading(t, aircraft, readOnlyStatus, aircraftOp)}</span>
          <span>{aircraftName}</span>
        </FormHeader>
        <FormBody>
          <PerfectScrollbar option={perfectScrollConfig}>
            <div>
              <h2>{t(`${sectionGeneral}.title`)}</h2>
              <Grid container spacing={3}>
                <Grid item xs={12} sm={12}>
                  <FormControl error={errors.aircraftType}>
                    <Input
                      inputProps={{
                        name: 'aircraftType',
                        maxLength: 4,
                      }}
                      className={errors.aircraftType ? 'error' : ''}
                      id="aircraftType"
                      label={t(`${sectionGeneral}.aircraftType`)}
                      color="secondary"
                      onChange={e => this.handleChange(e, '^[a-zA-Z0-9]+$')}
                      value={aircraftType}
                      error={errors.aircraftType}
                      required
                      disabled={readOnlyStatus}
                    />
                    <ErrorMessage
                      isVisible={errors.aircraftType}
                      message={codeErrorMessage}
                    />
                  </FormControl>
                </Grid>
                <Grid item xs={12} sm={12}>
                  <InputSelector
                    name="aircraftModel"
                    label={t(`${sectionGeneral}.aircraftModel`)}
                    required={true}
                    items={prepareModels(models)}
                    selected={aircraftModel}
                    handleChange={value =>
                      this.onSelectChange('aircraftModel', value)
                    }
                    disabled={readOnlyStatus}
                  />
                </Grid>
                <Grid item xs={12}>
                  <FormControl error={errors.aircraftName}>
                    <Input
                      inputProps={{
                        name: 'aircraftName',
                        maxLength: 50,
                      }}
                      id="aircraftName"
                      label={t(`${sectionGeneral}.aircraftName`)}
                      color="secondary"
                      className={errors.aircraftName ? 'error' : ''}
                      value={aircraftName}
                      required
                      error={errors.aircraftName}
                      onChange={e => this.handleChange(e, '^[a-zA-Z0-9].*')}
                      disabled={readOnlyStatus}
                    />
                    <ErrorMessage
                      isVisible={errors.aircraftName}
                      message={t(`${errorMsg}.aircraftName`)}
                    />
                  </FormControl>
                </Grid>
                <Grid item xs={12}>
                  <InputSelector
                    name="restFacility"
                    label={t(`${sectionGeneral}.restFacility`)}
                    required={true}
                    items={prepareRestFacilities(restFacilities)}
                    selected={restFacility}
                    handleChange={value =>
                      this.onSelectChange('restFacility', value)
                    }
                    disabled={readOnlyStatus}
                  />
                </Grid>
                <Grid item xs={12} sm={12}>
                  <h2>{t(`${sectionCrewComplement}.title`)}</h2>
                  <StandardCrewComplement
                    id="crewComposition"
                    t={t}
                    onChange={this.handleCrewComplements}
                    defaultValues={crewComposition}
                    openItemId={openItemId}
                    disabled={readOnlyStatus}
                  />
                </Grid>
              </Grid>
            </div>
          </PerfectScrollbar>
        </FormBody>
      </Form>
    );
  }
}

AircraftTypesForm.propTypes = {
  t: PropTypes.func.isRequired,
  openItemId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  aircraft: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    modelCode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    name: PropTypes.string,
    restFacilityCode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    code: PropTypes.string,
    crewComposition: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        quantity: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      })
    ),
  }),
  formId: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  handleCrewComplements: PropTypes.func.isRequired,
  models: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      code: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
  restFacilities: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
  inlineError: PropTypes.shape(),
  readOnly: PropTypes.bool.isRequired,
  enableReadOnly: PropTypes.bool,
};

AircraftTypesForm.defaultProps = {
  aircraft: {},
  inlineError: null,
  enableReadOnly: false,
};

export default AircraftTypesForm;
