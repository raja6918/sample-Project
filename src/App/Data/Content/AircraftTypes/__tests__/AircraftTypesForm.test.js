import React from 'react';
import { shallow } from 'enzyme';

import AircraftTypesForm from '../AircraftTypesForm';

const minProps = {
  aircraft: {
    id: 10,
    modelId: 10,
    name: 'Aircraft Test',
    restFacilityId: 'NONE',
    type: 'TEST',

    crewComposition: [
      {
        id: 1,
        quantity: 1,
      },
      {
        id: 2,
        quantity: 1,
      },
      {
        id: 5,
        quantity: 1,
      },
      {
        id: 12,
        quantity: 1,
      },
    ],
  },
  models: {
    data: [
      { id: 1, code: '777', name: 'test name' },
      { id: 2, code: '359', name: 'test name' },
    ],
  },
  restFacilities: [{ id: 1, name: 'C1-Bunk' }, { id: 2, name: 'C2-Recicling' }],
  formId: 'aircraftForm',
  isOpen: true,
  handleCrewComplements: jest.fn(),

  t: jest.fn(),
  position: {
    id: 10,
    name: 'Flyer',
    code: 'FLY',
    typeNam: 'Flight Deck',
    typeId: 1,
  },
  openItemId: '102',
  openItemName: 'My open item test',
  editMode: false,
  handleCancel: jest.fn(),
  handleOk: jest.fn(),
  onClose: jest.fn(),
};
const wrapper = shallow(<AircraftTypesForm {...minProps} />);

test('AircraftTypesForm Component renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('AircraftTypesForm Component renders correctly with an initial state', () => {
  expect(wrapper.instance().state).toMatchObject({
    aircraftName: '',
    aircraftType: '',
    aircraftModel: '',
    restFacility: 'NONE',
    errors: {
      aircraftName: false,
      aircraftType: false,
    },
    isFormDirty: false,
  });
});

test('AircraftTypesForm Executes its functions correctly and updates the state', () => {
  wrapper.instance().onSelectChange('aircraftModel', 3);
  expect(wrapper.state('aircraftModel')).toEqual(3);
  wrapper.instance().onSelectChange('restFacility', 2);
  expect(wrapper.state('restFacility')).toEqual(2);
  wrapper.instance().onFormChange();
  expect(wrapper.state('isFormDirty')).toEqual(false);
});

let readOnlyWrapper;
test('AircraftTypes Form component render correct in readOnly mode', () => {
  readOnlyWrapper = shallow(
    <AircraftTypesForm {...minProps} readOnly={true} />
  );
  expect(readOnlyWrapper).toMatchSnapshot();
});
