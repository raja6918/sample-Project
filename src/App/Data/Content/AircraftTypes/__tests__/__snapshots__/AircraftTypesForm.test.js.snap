// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AircraftTypes Form component render correct in readOnly mode 1`] = `
<Form
  anchor="right"
  cancelButton="Cancel"
  editMode={false}
  enableReadOnly={false}
  formId=""
  handleCancel={[MockFunction]}
  handleCrewComplements={[MockFunction]}
  handleOk={[MockFunction]}
  handleSubmit={[Function]}
  inlineError={null}
  isDisabled={false}
  isOpen={true}
  okButton="Ok"
  onChange={[Function]}
  onClose={[MockFunction]}
  openItemName="My open item test"
  position={
    Object {
      "code": "FLY",
      "id": 10,
      "name": "Flyer",
      "typeId": 1,
      "typeNam": "Flight Deck",
    }
  }
>
  <FormHeader>
    <span />
    <span />
  </FormHeader>
  <FormBody>
    <ScrollBar
      className=""
      component="div"
      containerRef={[Function]}
      option={
        Object {
          "swipeEasing": true,
          "wheelSpeed": 0.4,
        }
      }
    >
      <div>
        <h2 />
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <AircraftTypesForm__FormControl
              error={false}
            >
              <AircraftTypesForm__Input
                className=""
                color="secondary"
                disabled={true}
                error={false}
                id="aircraftType"
                inputProps={
                  Object {
                    "maxLength": 4,
                    "name": "aircraftType",
                  }
                }
                onChange={[Function]}
                required={true}
                value=""
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </AircraftTypesForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <InputSelector
              disabled={true}
              displayed="name"
              handleChange={[Function]}
              items={
                Array [
                  Object {
                    "id": "777",
                    "name": "test name",
                  },
                  Object {
                    "id": "359",
                    "name": "test name",
                  },
                ]
              }
              label=""
              name="aircraftModel"
              required={true}
              selected=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <AircraftTypesForm__FormControl
              error={false}
            >
              <AircraftTypesForm__Input
                className=""
                color="secondary"
                disabled={true}
                error={false}
                id="aircraftName"
                inputProps={
                  Object {
                    "maxLength": 50,
                    "name": "aircraftName",
                  }
                }
                onChange={[Function]}
                required={true}
                value=""
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </AircraftTypesForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <InputSelector
              disabled={true}
              displayed="name"
              handleChange={[Function]}
              items={
                Array [
                  Object {
                    "id": undefined,
                    "name": "C1-Bunk",
                  },
                  Object {
                    "id": undefined,
                    "name": "C2-Recicling",
                  },
                ]
              }
              label=""
              name="restFacility"
              required={true}
              selected="NONE"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <h2 />
            <StandardCrewComplement
              currentCrewGroupId={null}
              defaultValues={null}
              disabled={true}
              id="crewComposition"
              initialAssociatedPositionsValue={0}
              onChange={[Function]}
              openItemId="102"
              showOnlyAssociatedCGPositions={false}
              t={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "ERRORS.AIRCRAFT.aircraftType",
                    ],
                    Array [
                      "GLOBAL.form.add",
                    ],
                    Array [
                      "GLOBAL.form.save",
                    ],
                    Array [
                      "GLOBAL.form.close",
                    ],
                    Array [
                      "GLOBAL.form.cancel",
                    ],
                    Array [
                      "DATA.aircraft.form.edit",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.title",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftType",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftModel",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftName",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftName",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.restFacility",
                    ],
                    Array [
                      "DATA.aircraft.form.section.crewComplement.title",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftType",
                    ],
                    Array [
                      "GLOBAL.form.add",
                    ],
                    Array [
                      "GLOBAL.form.save",
                    ],
                    Array [
                      "GLOBAL.form.close",
                    ],
                    Array [
                      "GLOBAL.form.cancel",
                    ],
                    Array [
                      "DATA.aircraft.form.edit",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.title",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftType",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftModel",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftName",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftName",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.restFacility",
                    ],
                    Array [
                      "DATA.aircraft.form.section.crewComplement.title",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftType",
                    ],
                    Array [
                      "GLOBAL.form.add",
                    ],
                    Array [
                      "GLOBAL.form.save",
                    ],
                    Array [
                      "GLOBAL.form.close",
                    ],
                    Array [
                      "GLOBAL.form.cancel",
                    ],
                    Array [
                      "DATA.aircraft.form.edit",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.title",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftType",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftModel",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftName",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftName",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.restFacility",
                    ],
                    Array [
                      "DATA.aircraft.form.section.crewComplement.title",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftType",
                    ],
                    Array [
                      "GLOBAL.form.add",
                    ],
                    Array [
                      "GLOBAL.form.save",
                    ],
                    Array [
                      "GLOBAL.form.close",
                    ],
                    Array [
                      "GLOBAL.form.cancel",
                    ],
                    Array [
                      "DATA.aircraft.form.view",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.title",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftType",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftModel",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftName",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftName",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.restFacility",
                    ],
                    Array [
                      "DATA.aircraft.form.section.crewComplement.title",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </div>
    </ScrollBar>
  </FormBody>
</Form>
`;

exports[`AircraftTypesForm Component renders correctly 1`] = `
<Form
  anchor="right"
  cancelButton="Cancel"
  editMode={false}
  enableReadOnly={false}
  formId=""
  handleCancel={[MockFunction]}
  handleCrewComplements={[MockFunction]}
  handleOk={[MockFunction]}
  handleSubmit={[Function]}
  inlineError={null}
  isDisabled={true}
  isOpen={true}
  okButton="Ok"
  onChange={[Function]}
  onClose={[MockFunction]}
  openItemName="My open item test"
  position={
    Object {
      "code": "FLY",
      "id": 10,
      "name": "Flyer",
      "typeId": 1,
      "typeNam": "Flight Deck",
    }
  }
>
  <FormHeader>
    <span />
    <span />
  </FormHeader>
  <FormBody>
    <ScrollBar
      className=""
      component="div"
      containerRef={[Function]}
      option={
        Object {
          "swipeEasing": true,
          "wheelSpeed": 0.4,
        }
      }
    >
      <div>
        <h2 />
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <AircraftTypesForm__FormControl
              error={false}
            >
              <AircraftTypesForm__Input
                className=""
                color="secondary"
                disabled={false}
                error={false}
                id="aircraftType"
                inputProps={
                  Object {
                    "maxLength": 4,
                    "name": "aircraftType",
                  }
                }
                onChange={[Function]}
                required={true}
                value=""
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </AircraftTypesForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <InputSelector
              disabled={false}
              displayed="name"
              handleChange={[Function]}
              items={
                Array [
                  Object {
                    "id": "777",
                    "name": "test name",
                  },
                  Object {
                    "id": "359",
                    "name": "test name",
                  },
                ]
              }
              label=""
              name="aircraftModel"
              required={true}
              selected=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <AircraftTypesForm__FormControl
              error={false}
            >
              <AircraftTypesForm__Input
                className=""
                color="secondary"
                disabled={false}
                error={false}
                id="aircraftName"
                inputProps={
                  Object {
                    "maxLength": 50,
                    "name": "aircraftName",
                  }
                }
                onChange={[Function]}
                required={true}
                value=""
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </AircraftTypesForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <InputSelector
              disabled={false}
              displayed="name"
              handleChange={[Function]}
              items={
                Array [
                  Object {
                    "id": undefined,
                    "name": "C1-Bunk",
                  },
                  Object {
                    "id": undefined,
                    "name": "C2-Recicling",
                  },
                ]
              }
              label=""
              name="restFacility"
              required={true}
              selected="NONE"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <h2 />
            <StandardCrewComplement
              currentCrewGroupId={null}
              defaultValues={null}
              disabled={false}
              id="crewComposition"
              initialAssociatedPositionsValue={0}
              onChange={[Function]}
              openItemId="102"
              showOnlyAssociatedCGPositions={false}
              t={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "ERRORS.AIRCRAFT.aircraftType",
                    ],
                    Array [
                      "GLOBAL.form.add",
                    ],
                    Array [
                      "GLOBAL.form.save",
                    ],
                    Array [
                      "GLOBAL.form.close",
                    ],
                    Array [
                      "GLOBAL.form.cancel",
                    ],
                    Array [
                      "DATA.aircraft.form.edit",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.title",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftType",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftModel",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.aircraftName",
                    ],
                    Array [
                      "ERRORS.AIRCRAFT.aircraftName",
                    ],
                    Array [
                      "DATA.aircraft.form.section.general.restFacility",
                    ],
                    Array [
                      "DATA.aircraft.form.section.crewComplement.title",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </div>
    </ScrollBar>
  </FormBody>
</Form>
`;
